rllib/agents/mbmpo/mbmpo.py

```diff
@@ -0,0 +1,328 @@
+from ray.rllib.evaluation.metrics import collect_metrics
+
+logger = logging.getLogger(__name__)
+
+# yapf: disable
+# __sphinx_doc_begin__
+DEFAULT_CONFIG = with_common_config({
+    # If true, use the Generalized Advantage Estimator (GAE)
+    # with a value function, see https://arxiv.org/pdf/1506.02438.pdf.
+    "use_gae": True,
+    # GAE(lambda) parameter
+    "lambda": 1.0,
+    # Initial coefficient for KL divergence
+    "kl_coeff": 0.0005,
+    # Size of batches collected from each worker
+    "rollout_fragment_length": 200,
+    # Stepsize of SGD
+    "lr": 1e-3,
+    # Share layers for value function
+    "vf_share_layers": False,
+    # Coefficient of the value function loss
+    "vf_loss_coeff": 0.5,
+    # Coefficient of the entropy regularizer
+    "entropy_coeff": 0.0,
+    # PPO clip parameter
+    "clip_param": 0.5,
+    # Clip param for the value function. Note that this is sensitive to the
+    # scale of the rewards. If your expected V is large, increase this.
+    "vf_clip_param": 10.0,
+    # If specified, clip the global norm of gradients by this amount
+    "grad_clip": None,
+    # Target value for KL divergence
+    "kl_target": 0.01,
+    # Whether to rollout "complete_episodes" or "truncate_episodes"
+    "batch_mode": "complete_episodes",
+    # Which observation filter to apply to the observation
+    "observation_filter": "NoFilter",
+    # Number of Inner adaptation steps for the MAML algorithm
+    "inner_adaptation_steps": 1,
+    # Number of MAML steps per meta-update iteration (PPO steps)
+    "maml_optimizer_steps": 8,
+    # Inner Adaptation Step size
+    "inner_lr": 1e-3,
+    # Horizon of Environment (200 in MB-MPO paper)
+    "horizon": 200,
+    # Dynamics Ensemble Hyperparameters
+    "dynamics_model": {
+        "custom_model": DynamicsEnsembleCustomModel,
+        # Number of Transition-Dynamics Models for Ensemble
+        "model_ensemble_size": 1,
+        # Hidden Layers for Model Ensemble
+        "model_hiddens": [512, 512],
+        # Model Learning Rate
+        "model_lr": 1e-3,
+        # Max number of training epochs per MBMPO iter
+        "model_train_epochs": 1,
+        # Model Batch Size
+        "model_batch_size": 500,
+        # Training/Validation Split
+        "valid_split_ratio": 0.2,
+        # Normalize Data (obs, action, and deltas)
+        "normalize_data": True,
+    },
+    # Workers sample from dynamics models
+    "custom_vector_env": custom_model_vector_env,
+    # How many enviornments there are per worker (vectorized)
+    "num_worker_envs": 50,
+    # How many iterations through MAML per MBMPO iteration
+    "num_maml_steps": 5,
+})
+# __sphinx_doc_end__
+# yapf: enable
+
+# Select Metric Keys for MAML Stats Tracing
+METRICS_KEYS = [
+    "episode_reward_mean", "episode_reward_min", "episode_reward_max"
+]
+
+
+class MetaUpdate:
+    def __init__(self, workers, num_steps, maml_steps, metric_gen):
```

Comment:
Please add a detailed docstring as to what this exec-related method does and which args it takes in the c'tor.